app/agents_v2/primary_agent/llm_registry.py

```
@@ -0,0 +1,124 @@
+        SupportedModel enum value
+        
+    Raises:
+        ValueError: If the model ID is not supported
+    """
+    try:
+        return SupportedModel(model_id)
+    except ValueError:
+        raise ValueError(
+            f"Unsupported model: {model_id}. "
+            f"Supported models are: {', '.join(m.value for m in SupportedModel)}"
+        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)